File: Front-end/src/app/pages/gestion-marketplace/models/Commande.ts
import { Status } from "./Status";
import { LigneCommande } from "./LigneCommande";
import { Livraison } from "./Livraison";

export class Commande {
    idCommande?: number;
    dateCommande: Date;
    montantCommande: number;
    nom: string;
    phone: number;
    email: string;
    adresse: string;
    gouvernement: string;
    city: string;
    shippingMethod: string;
    AdditionalService: number;
    ligneCommande!: LigneCommande[];
    userId!: number;
    etat!: Status;
    OrderNumber!: string;
    livraison?: Livraison;
    paymentMethod?: string;

    constructor() {
        this.dateCommande = new Date();
      //  this.ligneCommande = [];
        this.montantCommande = 0;
        this.nom = '';
        this.phone = 0;
        this.email = '';
        this.adresse = '';
        this.gouvernement = '';
        this.city = '';
        this.shippingMethod = '';
        this.AdditionalService = 0;
        this.etat=Status.IN_PROGRESS;
        this.OrderNumber='';

    }
}

File: Back-end Microservice/MarketPlace-Service/src/main/java/tn/esprit/spring/marketplaceservice/entity/Commande.java
package tn.esprit.spring.marketplaceservice.entity;


import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDate;
import java.util.List;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PUBLIC)
public class Commande {

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        Long idCommande;
        LocalDate dateCommande;
        Double montantCommande;
        String nom;
        Long phone;
        String email;
        String adresse;
        String gouvernement;
        String city;
        String shippingMethod;
        Long AdditionalService;
        Long userId;
        @Enumerated(EnumType.STRING)
        Status etat;
        String OrderNumber;



        @JsonIgnore
        @OneToMany(mappedBy = "commande")
        private List<LigneCommande> ligneCommande;

        @JsonIgnore
        @ManyToOne
        Livraison livraison;
}
